pkg/helper/workloads.go

```diff
@@ -0,0 +1,152 @@
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// ApplyYamlInFolder reads a folder and attempts to create objects in K8s with the yaml
+func ApplyYamlInFolder(folder, namespace string, kube kubernetes.Interface) ([]runtime.Object, error) {
+	var (
+		objects []runtime.Object
+		obj     runtime.Object
+		files   []string
+		err     error
+	)
+	pwd, _ := os.Getwd()
+	err = filepath.Walk(pwd+folder, func(path string, info os.FileInfo, err error) error {
+		if strings.Contains(path, ".yaml") || strings.Contains(path, ".yml") {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return objects, err
+	}
+
+	for _, file := range files {
+		if obj, err = ReadK8sYaml(file); err != nil {
+			return objects, err
+		}
+		if obj, err = CreateRuntimeObject(obj, namespace, kube); err != nil {
+			return objects, err
+		}
+		objects = append(objects, obj)
+	}
+
+	return objects, nil
+}
+
+// ReadK8sYaml reads a file at the specified path and attempts to decode it into a runtime.Object
+func ReadK8sYaml(file string) (runtime.Object, error) {
+	var (
+		absfile string
+		err     error
+	)
+
+	if absfile, err = filepath.Abs(file); err != nil {
+		return nil, err
+	}
+
+	f, err := ioutil.ReadFile(absfile)
+	if err != nil {
+		return nil, err
+	}
+
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	obj, _, err := decode([]byte(f), nil, nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error while decoding YAML object. Err was: %s", err)
+	}
+
+	return obj, nil
+}
+
+// CreateRuntimeObject takes a runtime.Object and attempts to create an object in K8s with it
+func CreateRuntimeObject(obj runtime.Object, ns string, kube kubernetes.Interface) (runtime.Object, error) {
+	var (
+		newObj runtime.Object
+		ok     bool
+		err    error
+	)
+	if obj == nil {
+		return nil, fmt.Errorf("Nil object passed in")
+	}
+
+	if _, err := kube.CoreV1().Namespaces().Get(ns, metav1.GetOptions{}); err != nil {
+		_, err := kube.CoreV1().Namespaces().Create(&corev1.Namespace{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "namespace",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ns,
+			},
+			Spec: corev1.NamespaceSpec{},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Error creating namespace: %s", err.Error())
+		}
+
+		// Need to wait till the namespace is actually created/active before proceeding
+		// Namespace creation is fairly swift, so these numbers are arbitrary.
+		// If this times out, there's something wrong.
+		wait.PollImmediate(2*time.Second, 1*time.Minute, func() (bool, error) {
+			if _, err := kube.CoreV1().Namespaces().Get(ns, metav1.GetOptions{}); err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+	}
+
+	switch obj.GetObjectKind().GroupVersionKind().Kind {
```

Comment:
This is the only thing I don't like.

Currently we only support the below-defined K8s objects. The way to do dynamic objects is hacky and frankly more pain than it's worth. 

If additional workloads are added that don't fall under these objects we will need to add support here as well.

Reply:
:+1:

Reply:
Let's add a TODO here to convey that for future implementers.